lib/perl/Genome/Model/Tools/Bedpe/EvaluateBedpe.pm

```diff
@@ -67,10 +77,84 @@ sub _get_stat {
         input_file_b => $file_b,
         intersection_type => $type,
     );
+    return $output_file;
+}
 
+sub _get_false_negatives {
+    my $self = shift;
+    my $output_file = $self->_get_pair_to_pair_output($self->gold_bedpe, $self->bedpe, 'notboth');
     return $self->_unique_sv_count($output_file);
 }
 
+sub _get_true_positives {
+    my $self = shift;
+    my $output_file = $self->_get_pair_to_pair_output($self->bedpe, $self->gold_bedpe, 'both');
+    return $self->_sv_with_min_support_count($output_file, $self->bedpe, $self->min_hit_support, $self->true_positive_file);
+}
+
+sub _sv_with_min_support_count {
+    my ($self, $file, $original_file, $min_hit_support, $output) = @_;
+    my $f = new IO::File($file, "r");
+    my %results;
+
+    my $id_index = _count_cols($original_file) + 6;
+
+    my $out;
+    if ($output) {
+        $out = Genome::Sys->open_file_for_writing($output);
+    }
+    while (my $line = $f->getline) {
+        chomp $line;
+        my @f = split("\t", $line);
+        my $sv_name = $f[6];
+        my $read_id = $f[$id_index];
+        unless (defined $sv_name) {
+            die "Sv name not defined on line:\n$line";
+        }
+        unless (defined $read_id) {
+            die "read id not defined on line:\n$line";
+        }
+        ++$results{$sv_name}{count} unless exists $results{$sv_name}{reads}{$read_id};
+        $results{$sv_name}{reads}{$read_id} = 1;
+        $results{$sv_name}{sv} = \@f;
+    }
+    $f->close();
+
+    my $in = new IO::File($original_file, "r");
+
+    my %seen;
+    my $count = 0;
+    while (my $line = $in->getline) {
+        chomp $line;
+        my @f = split("\t", $line);
+        my $sv_name = $f[6];
+        unless (defined $sv_name) {
+            die "Sv name not defined on line:\n$line";
+        }
+        next if exists $seen{$sv_name};
+        $seen{$sv_name} = 1;
+
+        next unless exists $results{$sv_name} && ($results{$sv_name}{count} >= $min_hit_support);
+        $count++;
+        if ($output) {
```

Comment:
It doesn't make a big difference, but would it be better for this to check `if ($out)`?